gherkin/javascript/src/legacy/gherkin/generate_events.js

```diff
@@ -17,55 +35,50 @@ function generateEvents(data, uri, types, language) {
   try {
     if (types['source']) {
       result.push({
-        type: 'source',
-        uri: uri,
-        data: data,
-        media: {
-          encoding: 'utf-8',
-          type: 'text/x.cucumber.gherkin+plain'
+        source: {
+          uri: uri,
+          data: data,
+          media: {
+            encoding: 'UTF8',
+            contentType: 'text/x.cucumber.gherkin+plain'
+          }
         }
       })
     }
 
     if (!types['gherkin-document'] && !types['pickle'])
       return result
 
-    var gherkinDocument = parser.parse(data, language)
+    var gherkinDocument = parser.parse(data, new TokenMatcher(language))
```

Comment:
Btw, notice this change. It turns out that some of the tests for the [EventStream](https://github.com/cucumber/cucumber/blob/gherkin-v5.1.0/gherkin/javascript/test/stream/event_stream_test.js) (v5.1.0) weren't actually run as part of CI. Hence this probably went unnoticed.